agnpy/absorption/absorption.py

```diff
@@ -479,6 +571,62 @@ def tau_blr_mu_s(self, nu):
             phi=self.phi,
         )
 
+    def tau_blr_cubepy(self, nu, eps_abs=1e-6):
+        """Evaluates the gamma-gamma absorption produced by a spherical shell
+        BLR for a general set of model parameters with cubepy integration
+        method
+        """
+
+        tau_blr_list = []
+        for freq in nu:
+            tau_blr_list.append(
+            self.evaluate_tau_blr_cubepy(
+                freq,
+                eps_abs,
+                self.z,
+                self.mu_s,
+                self.target.L_disk,
+                self.target.xi_line,
+                self.target.epsilon_line,
+                self.target.R_line,
+                self.r,
+                mu=self.mu,
+                phi=self.phi,
+                            ))
+        return tau_blr_list
+
+    def tau_blr_all_lines_cubepy(self, nu, eps_abs=1e-6):
+        """
+        Calculate the absorption in all available BLR (Broad Line Region) shells.
+        The radius and luminosity are normalized to Hbeta.
+
+        Parameters
+        ----------
+        nu: numpy array
+            Frequency array with unit
+        eps_abs: float, optional
+            Absolute precision for the calculation, default is 1e-6
+        """
+        SUM_RATIO_LINES_BLR = 30.652
+        XI_TARGET_LINE = self.target.xi_line
+        tau_z_all = np.zeros(nu.shape)
+
+        blr_shells = dict()
+        absorption_shells = dict()
+
+        for line_key in lines_dictionary.keys():
```

Comment:
I am a bit lost with the calculations in this `for` loop.
Are you assuming that the `target` is a `SphericalShellBLR` emitting the `Hbeta` line? If this is not the case then I have troubles in understanding why you scale radius and luminosity to the current (arbitrary) line.

Reply:
If, on the other hand, the user has to use the `Hbeta` line as a target, then this should be stated explicitly somewhere.

Reply:
Hbeta is used as a reference line for radius and luminosity scaling to ensure consistency in calculations for different BLR lines.

Reply:
I think this is solved already (the explanation is in the docstring of the function), resolving..